sdk/appplatform/mgmt/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppDeploymentImpl.java

```diff
@@ -0,0 +1,446 @@
+                        tarArchiveOutputStream.putArchiveEntry(entry);
+                        IOUtils.copy(inputStream, tarArchiveOutputStream);
+                        tarArchiveOutputStream.closeArchiveEntry();
+                    }
+                } else {
+                    tarArchiveOutputStream.putArchiveEntry(entry);
+                    tarArchiveOutputStream.closeArchiveEntry();
+                }
+            }
+        }
+        return compressFile;
+    }
+
+    private Mono<ShareFileAsyncClient> createShareFileAsyncClient(ResourceUploadDefinition option, long maxSize) {
+        ShareFileAsyncClient shareFileAsyncClient = new ShareFileClientBuilder()
+            .endpoint(option.uploadUrl())
+            .httpClient(manager().httpPipeline().getHttpClient())
+            .buildFileAsyncClient();
+
+        return shareFileAsyncClient.create(maxSize)
+            .then(Mono.just(shareFileAsyncClient));
+    }
+
+    private Mono<ShareFileAsyncClient> uploadToStorage(byte[] bytes, ResourceUploadDefinition option) {
+        inner().properties().source().withRelativePath(option.relativePath());
+        return createShareFileAsyncClient(option, bytes.length)
+                .flatMap(shareFileAsyncClient -> {
+                    List<Integer> blockList = new ArrayList<>();
+                    for (int start = 0; start < bytes.length; start += BLOCK_SIZE) {
+                        blockList.add(start);
+                    }
+                    return Flux.fromIterable(blockList)
+                        .flatMap(start -> {
+                            int length = Math.min(bytes.length - start, BLOCK_SIZE);
+                            return shareFileAsyncClient.uploadWithResponse(
+                                Flux.just(ByteBuffer.wrap(bytes, start, length)), length, (long) start);
+                        })
+                        .then(Mono.just(shareFileAsyncClient));
+                });
+    }
+
+    private Mono<ShareFileAsyncClient> uploadToStorage(File source, ResourceUploadDefinition option) {
+        inner().properties().source().withRelativePath(option.relativePath());
+        try {
+            return createShareFileAsyncClient(option, source.length())
+                .flatMap(shareFileAsyncClient -> shareFileAsyncClient.uploadFromFile(source.getAbsolutePath())
+                    .then(Mono.just(shareFileAsyncClient)));
+        } catch (Exception e) {
+            return Mono.error(e);
+        }
+    }
+
+    @Override
+    public SpringAppDeploymentImpl withJarPath(File jar) {
+        ensureSource();
+        inner().properties().source().withType(UserSourceType.JAR);
+        this.addDependency(
+            context -> parent().getResourceUploadUrlAsync()
+                .flatMap(option -> uploadToStorage(jar, option)
+                    .then(context.voidMono()))
+        );
+        return this;
+    }
+
+    @Override
+    public SpringAppDeploymentImpl withJarFile(byte[] jar) {
```

Comment:
Should we provide `byte[]` as input?

Reply:
Just remove this function? only provider file as input?

Reply:
I found these functions name should be changed.

Reply:
> I found these functions name should be changed.

Reason?

Reply:
I firstly use Path and byte[] for the jarPath and jarFile, but later I change parameter to File, which I think the function should also be changed.

Reply:
removed the `byte[]` function